src/Components/Auth/Login.tsx

```diff
@@ -109,7 +109,13 @@ export const Login = (props: { forgot?: boolean }) => {
           window.location.pathname === "/" ||
           window.location.pathname === "/login"
         ) {
-          window.location.href = "/facility";
+          const lastPath = localStorage.getItem("lastPath");
+          if (lastPath) {
+            localStorage.removeItem("lastPath");
+            window.location.href = lastPath;
+          } else {
+            window.location.href = "/facility";
+          }
```

Comment:
Use query params instead of local storage.

something like, `/session-expired?redirect=<url encoded url of the last url>`